flutter/lib/src/default_integrations.dart

```diff
@@ -370,15 +370,27 @@ class LoadReleaseIntegration extends Integration<SentryFlutterOptions> {
         if (name.isEmpty) {
           // Not all platforms have a packageName.
           // If no packageName is available, use the appName instead.
-          name = _cleanAppName(packageInfo.appName);
+          name = _cleanString(packageInfo.appName);
+        }
+
+        final version = _cleanString(packageInfo.version);
```

Comment:
This block is guarded by _either_ `release` or `dist` being empty on options. But below we only take the `release` built in here, if `options.release` is not empty, we just use that, and same for `dist`. 

So if only `options.release` is set, we'll build a new release (as per this block) but won't use it at all.
Would it make sense to check if `+` exists and append the `buildNumber` we detected in that case?

Reply:
I think the `??` does the job if I got what you meant eg

`options.release = options.release ?? release;`

but what we could do is to actually check if we need to build the `release` or `dist` only and not execute code that is not necessary, eg, building a new `release` if we're going to use the `options.release` anyway as its not empty

Reply:
yeah it's already using options.release if it's already set, it's just odd that we build a new release value even though it's not used.

And I was trying to think of ways to combine what we built in the block with what was defined but best not mess around with a value that was explicitly set by the user on options.

So the only improvement here is as you said just to avoid doing the work if not needed though it'll make the logic more complicated